drivers: Use of proper base address macro across all ambiq platforms

RichardSWheatley opened the pull request:
REG_X_BASEADDR will be removed from all hal files.
This forces the use of the peripheral base address

swift-tk commented:
We need a private macro to obtain the port in memc_mspi_aps6404l.c and flash_mspi_atxp032.c since these device drivers are meant to be shared. I had clearly missed it in previous PR.
@RichardSWheatley Would you like to fix it?

gmarull reviewed (dismissed):
please, make commit message better. Title is misleading.

RichardSWheatley commented:
> We need a private macro to obtain the port in memc_mspi_aps6404l.c and flash_mspi_atxp032.c since these device drivers are meant to be shared. I had clearly missed it in previous PR. @RichardSWheatley Would you like to fix it?

I can create a ticket and work on it internally, yes.

swift-tk commented:
> > We need a private macro to obtain the port in memc_mspi_aps6404l.c and flash_mspi_atxp032.c since these device drivers are meant to be shared. I had clearly missed it in previous PR. @RichardSWheatley Would you like to fix it?
> 
> I can create a ticket and work on it internally, yes.

I was hoping you could do it all together in this PR since you are changing the files.

Something like this, private as in platform specific to ambiq but not in hal. Or move the entire section of `CONFIG_SOC_FAMILY_AMBIQ` to mspi_ambiq.h so that it looks cleaner.
![image](https://github.com/user-attachments/assets/1df8ee1a-6f11-46ac-96f9-8d46fad0ba53)

swift-tk commented on drivers/flash/flash_mspi_atxp032.c:
```diff
@@ -826,7 +826,7 @@ static const struct flash_driver_api flash_mspi_atxp032_api = {
 #define FLASH_MSPI_ATXP032(n)                                                                     \
 	static const struct flash_mspi_atxp032_config flash_mspi_atxp032_config_##n = {           \
-		.port = (DT_REG_ADDR(DT_INST_BUS(n)) - REG_MSPI_BASEADDR) /                       \
+		.port = (DT_REG_ADDR(DT_INST_BUS(n)) - MSPI0_BASE) /                       \
```
Please don't forget to align the slash on the right.

swift-tk commented on drivers/flash/flash_mspi_atxp032.c:
```diff
@@ -826,9 +826,8 @@ static const struct flash_driver_api flash_mspi_atxp032_api = {
-			(DT_REG_SIZE(DT_INST_BUS(n)) * 4),                                        \
-		.mem_size = DT_INST_PROP(n, size) / 8,                                            \
+		.port        = MSPI_PORT(n),                                                   \
+		.mem_size    = DT_INST_PROP(n, size) / 8,                                      \
```
The slashes are still misaligned.

swift-tk commented on drivers/mspi/mspi_ambiq.h:
```diff
@@ -47,6 +47,9 @@ enum mspi_ambiq_timing_param {
 	MSPI_AMBIQ_SET_RXDQSDLYEXT    = BIT(7),
 };
 
+#define MSPI_PORT(n)    ((DT_REG_ADDR(DT_INST_BUS(n)) - MSPI0_BASE) /                    \
```
Align slash to the 100th column.

swift-tk commented:
The rest LGTM

RichardSWheatley commented:
> please, make commit message better. Title is misleading.

@gmarull please review the changes you requested

gmarull reviewed (commented):
https://github.com/zephyrproject-rtos/zephyr/pull/76707/commits/7079f166fb9b8e02dde1030d3f2bba664d5d9aca Commit title is still unclear/too generic

RichardSWheatley commented:
> [7079f16](https://github.com/zephyrproject-rtos/zephyr/commit/7079f166fb9b8e02dde1030d3f2bba664d5d9aca) Commit title is still unclear/too generic

It wasn't changed to this by me. It was changed to this by the assignee.

gmarull commented:
> > [7079f16](https://github.com/zephyrproject-rtos/zephyr/commit/7079f166fb9b8e02dde1030d3f2bba664d5d9aca) Commit title is still unclear/too generic
> 
> It wasn't changed to this by me. It was changed to this by the assignee.

Commit title is still unclear/too generic

gmarull reviewed (dismissed):
Please, do not dismiss change requests which are clear and explicit.

RichardSWheatley commented:
> Please, do not dismiss change requests which are clear and explicit.

They were not and that is why they were dismissed.

"Use of proper base address macro across all ambiq platforms" is the exact and only reason for the change.
If you do not think that is enough, then suggest something else. If not, I will dismiss again.

gmarull commented:
> > Please, do not dismiss change requests which are clear and explicit.
> 
> They were not and that is why they were dismissed.
> 
> "Use of proper base address macro across all ambiq platforms" is the exact and only reason for the change. If you do not think that is enough, then suggest something else. If not, I will dismiss again.

Isn't this sufficient?
> [7079f16](https://github.com/zephyrproject-rtos/zephyr/commit/7079f166fb9b8e02dde1030d3f2bba664d5d9aca) Commit title is still unclear/too generic

RichardSWheatley commented:
> > > Please, do not dismiss change requests which are clear and explicit.
> > 
> > 
> > They were not and that is why they were dismissed.
> > "Use of proper base address macro across all ambiq platforms" is the exact and only reason for the change. If you do not think that is enough, then suggest something else. If not, I will dismiss again.
> 
> Isn't this sufficient?
> 
> > [7079f16](https://github.com/zephyrproject-rtos/zephyr/commit/7079f166fb9b8e02dde1030d3f2bba664d5d9aca) Commit title is still unclear/too generic

Yes, my title is sufficient.

gmarull commented:
> > > > Please, do not dismiss change requests which are clear and explicit.
> > > 
> > > 
> > > They were not and that is why they were dismissed.
> > > "Use of proper base address macro across all ambiq platforms" is the exact and only reason for the change. If you do not think that is enough, then suggest something else. If not, I will dismiss again.
> > 
> > 
> > Isn't this sufficient?
> > > [7079f16](https://github.com/zephyrproject-rtos/zephyr/commit/7079f166fb9b8e02dde1030d3f2bba664d5d9aca) Commit title is still unclear/too generic
> 
> Yes, my title is sufficient.

I guess you're confused about commit and PR titles, which are not the same thing. Finding `drivers: update to use proper Base Address` in git history is just useless, please, make it more clear.

RichardSWheatley commented:
> drivers: update to use proper Base Address

And yet that is exactly what this PR addresses, so what is your suggestion?

RichardSWheatley committed:
drivers: update AMBIQ drivers to use proper base address

REG_X_BASEADDR will be removed from all hal files.
This forces the use of the peripheral base address
Define MSPI_PORT macro for chip drivers

Signed-off-by: Richard Wheatley <richard.wheatley@ambiq.com>

RichardSWheatley committed:
drivers: spi: remove old ambiq mspi

Remove the old Ambiq MSPI

Signed-off-by: Richard Wheatley <richard.wheatley@ambiq.com>

RichardSWheatley commented:
The video showed me nothing more than [the](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines) which also adds nothing more than the original comment.

What did help was this "are you then updating all drivers in-tree to use "proper" (?) base addresses?"

Had you mentioned that originally, I would have completely understood what you meant in the first place.
